Tools/c-analyzer/c_parser/info.py

```diff
@@ -63,18 +89,17 @@ def validate(self):
         """Fail if the object is invalid (i.e. init with bad data)."""
         self._validate_id()
 
+        if self.storage is None or self.storage == info.UNKNOWN:
+            raise TypeError('missing storage')
+        elif self.storage not in self.STORAGE:
+            raise ValueError(f'unsupported storage {self.storage:r}')
+
         if self.vartype is None or self.vartype == info.UNKNOWN:
             raise TypeError('missing vartype')
 
     @property
     def isglobal(self):
-        try:
-            return self._isglobal
-        except AttributeError:
-            # XXX Include extern variables.
-            # XXX Ignore functions.
-            self._isglobal = ('static' in self.vartype.split())
-            return self._isglobal
+        return self.storage != 'local'
```

Comment:
Wouldn't an implicit also not be considered a global?

```suggestion
        return self.storage != 'local' and self.storage != 'implicit'
```

Reply:
In the context of the tool, "global" is any variable that maintains state at a process-global level.  That includes non-local implicits.